syntax = "proto2";
package p2pnet.protocol;

message p2pMessageHeader {
	optional bytes src_id = 1;
	optional bytes dest_id = 2;
}

message p2pMessage {
	required p2pMessageHeader message_header = 1;
	enum MessageType {
		DATA = 0;
		ONION = 1;
		AGREEMENT = 2;
	}

	optional MessageType message_type = 2 [default = DATA];
	optional bytes message_s = 3;
}

message p2pMessage_Agreement {
	required bytes src_pubkey = 1;	// My own public key in BER format
}